packages/editor/src/components/local-autosave-monitor/index.js

```diff
@@ -0,0 +1,191 @@
+/**
+ * External dependencies
+ */
+import { once, uniqueId, omit } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useEffect, useRef } from '@wordpress/element';
+import { ifCondition } from '@wordpress/compose';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { parse } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import AutosaveMonitor from '../autosave-monitor';
+
+const requestIdleCallback = window.requestIdleCallback ? window.requestIdleCallback : window.requestAnimationFrame;
+
+/**
+ * Time interval at which local autosave should occur after a change, in
+ * seconds.
+ *
+ * @var {number}
+ */
+const AUTOSAVE_INTERVAL_SECONDS = 15;
+
+/**
+ * Function which returns true if the current environment supports browser
+ * sessionStorage, or false otherwise. The result of this function is cached and
+ * reused in subsequent invocations.
+ */
+const hasSessionStorageSupport = once( () => {
+	try {
+		// Private Browsing in Safari 10 and earlier will throw an error when
+		// attempting to set into sessionStorage. The test here is intentional in
+		// causing a thrown error as condition bailing from local autosave.
+		window.sessionStorage.setItem( '__wpEditorTestSessionStorage', '' );
+		window.sessionStorage.removeItem( '__wpEditorTestSessionStorage' );
+		return true;
+	} catch ( error ) {
+		return false;
+	}
+} );
+
+/**
+ * Function returning a sessionStorage key to set or retrieve a given post's
+ * automatic session backup.
+ *
+ * Keys are crucially prefixed with 'wp-autosave-' so that wp-login.php's
+ * `loggedout` handler can clear sessionStorage of any user-private content.
+ *
+ * @see https://github.com/WordPress/wordpress-develop/blob/6dad32d2aed47e6c0cf2aee8410645f6d7aba6bd/src/wp-login.php#L103
+ *
+ * @param {string} postId  Post ID.
+ * @return {string}        sessionStorage key
+ */
+function postKey( postId ) {
+	return `wp-autosave-block-editor-post-${ postId }`;
+}
+
+/**
+ * Custom hook which returns a callback function to be invoked when a local
+ * autosave should occur.
+ *
+ * @return {Function} Callback function.
+ */
+function useAutosaveCallback() {
+	const {
+		postId,
+		getEditedPostAttribute,
+	} = useSelect( ( select ) => ( {
+		postId: select( 'core/editor' ).getCurrentPostId(),
+		getEditedPostAttribute: select( 'core/editor' ).getEditedPostAttribute,
+	} ) );
+
+	return useCallback( () => {
+		const saveToSessionStorage = () => {
+			window.sessionStorage.setItem( postKey( postId ), JSON.stringify( {
+				post_title: getEditedPostAttribute( 'title' ),
```

Comment:
What about other stuff like:
- taxonomy
- categories
- featured image

and so on?

Reply:
Worth noting that this wasn't saved in the classic editor, but that doesn't mean we should do the same. Just noting.

Reply:
Yeah. Grzegorz and I also talked a little about this:

* There's also a management of expectations to play with: the more data we autosave, the greater the expectations that this is a fully fledged offline support feature, which it isn't.

* I think backing up post content really is 90% of the feature, and for an introductory PR this is more than fine. The fact that Classic did the same validates this.

* As Gutenberg evolves to support more than just posts with content and into the realm of full-site editing, a lot of this might become a lot more ambitious if we're dealing with multiple entities to save in a single editor. (When we get there we can try autosaving around each `BlockEditor` and see where it takes us.)

For now, let's keep it focused on these three bits.